geopandas/geodataframe.py

```diff
@@ -589,7 +647,7 @@ def __getitem__(self, key):
         geo_col = self._geometry_column_name
         if isinstance(result, Series) and isinstance(result.dtype, GeometryDtype):
             result.__class__ = GeoSeries
-            result.crs = self.crs
+            result.crs = result.values.crs
```

Comment:
It might be that this line is not needed anymore, since the crs is stored on the array, and the returned Series still has a GeometryArray under the hood (since we check the dtype to be GeometryDtype)